exec/execution.go

```diff
@@ -6,16 +6,18 @@ import (
 )
 
 type Execution struct {
-	start    time.Time
-	end      time.Time
-	jobTitle string
-	err      error
+	start         time.Time
+	end           time.Time
+	jobTitle      string
+	err           error
+	correlationId string
 }
 
-func StartExecution(jobTitle string) *Execution {
+func StartExecution(jobTitle string, correlationId string) *Execution {
```

Comment:
At this point, you could hold a reference to the starting context instead of a copy of the context's correlationId. With the Execution, you would use the context's CorrelationId()-Getter. 
This could be useful in future development.